### ISSUE TITLE
Accept XLIFF as a source file format

### ISSUE THREAD

**Issue by shghaznawi** [heartsome/translationstudio8#11]
This is a great tool with a lot of potential. We want it to accept XLIFF file format generated by other tools and TMS's. This make this tool more appealing with this type of function. I believe it should be fairly easy to implement since it is an XLIFF editor and marketed as such.

I would be happy to help with this feature and spec'ing out the requirements.

**Comment by Alf71** [comment]
Please read the manuals available somewhere in the packages or in Internet.
You just need to change the extension generated by the respective CAT (memo xlif, trados xlif) into Studio 8 normal XLIFF extension

**Comment by shghaznawi** [comment]
Thanks for such a quick response. Does it mean I can import the XLIFF after extension change or it will be opened as a Heartsome project file?

**Comment by Alf71** [comment]
... it will be opened as a Heartsome project file.

Heartsome is compatible with the following translation tools' native file formats.
SDL TRADOS FILES
Heartsome supports pre-segmented TTX files, Trados Uncleaned Word files and SDLXLIFF files.
1. To pre-segment TTX files in SDL Trados Workbench
a. Open SDL Trados Workbench.
b. Select Tools > Translate from the menu bar, put a check in the "Segment unknown sentences" box.
c. Click the Add button to add the files for processing or use the mouse to drag-and-drop files to the "Files to translate" field.
d. Click the Translate button.
2. Then convert the TTX files directly in Heartsome, just like other ordinary files.
3. After translation, convert *.ttx.hsxliff to the target language files.
4. In the target language folder under the Heartsome project, copy-and-paste the TTX to the local disk, and then return them to the project owner.
MEMOQ 6.X FILES
Export MQXLZ from MemoQ, then convert MQXLZ files just like ordinary files directly in a Heartsome.
DÉJÀ VU 2 FILES
Export the XLIFF files from Déjà Vu 2, then process the XLIFF in Heartsome.
To export XLIFF from Déjà Vu 2:
1. From the menu bar, choose File > Export> External View.
2. Select the storage path in the Destination field.
3. Select XLIFF under Options in the same dialog box.
4. Click OK.

WORDFAST TXML FILES
TXML is the native file format of Wordfast, which is XML based. You can process TXML files directly in Heartsome, just like other ordinary files.

**Comment by shghaznawi** [comment]
Again, thanks for the response. I am trying to get this to work. I have GlobalSight XLIFF 1.2 file that I need to test for translation with Heartsome. Does it mean I have to change the extension to .hsxliff?

Also, does Heartsome allow for integration with GlobalSight TMS?

**Comment by Alf71** [comment]
I think GS XLIFF is handled a bit different but you can try it.
And please don't mix everything in the same bag.
GS is GS and Heartsome is Heartsome

**Comment by shghaznawi** [comment]
Hi again,

Thanks but isn't the purpose of a standard like XLIFF that it will be cross-compatible? I am not mixing GS with HS, I am trying to see if I can get them to work together so I can offer that as a solution to someone interested.

That being said, a tool that calls itself standards based is not so much standards based if it supports its own flavour of standard which is basically proprietary.

**Comment by Alf71** [comment]
GS has a different XLIF approach

**Comment by cburatto** [comment]
This is an issue not restricted to HS. Every tool implements their own version of Xliff. That's why you need to convert SDLXLIFF files into HSXLIFF format. This might change with Xliff 2.0.

However, if the Heartsome OS development does not continue, it will quickly be outdated. Not only xliff compatibility needs to be kept, but there is the need to keep converters to other file formats up to date.

**Comment by shghaznawi** [comment]
True, a lot of effort and work has gone into HS and after spending half an hour with it I believe it has a lot of potential to compete with OmegaT or even beyond that. The only issue is that it should at least offer compatibility with other open source tools that are complementing, not competing with it. GS being a good example, I would recommend HS to a large number of my customers if I find a way to make it work with GS working files.

I understand that this situation is not unique to HS and every tool has its own flavour of the "standard". But that defeats the purpose of the standards if every tool creates its own standard on top of the standard for competitive reasons.

**Comment by shghaznawi** [comment]
@Alf71 I guess there is no point evaluating the tool as you mentioned it doesn't like GS type of XLIFF. Does it need a lot of work to make HS read XLIFF 1.2 generated by GS?

**Comment by Alf71** [comment]
Segmentate/Translate DOC with HS8 and export TMX into GS...
Both navigations are quite inflexible, but I would say GS is worst than HS8 (regarding the translation navigation issues)
